syntax = "proto3";

service rollAttack {
	rpc rollAttack(rollAttackmsg) returns (rollReply) {}
}

service rollInitiative {
	rpc rollInitiative(rollIniativemsg) returns (rollReply) {}
}

service rollSpellcast {
	rpc rollSpellcast(rollSpellcastmsg) returns (rollReply) {}
}

service rollSpellDamage {
	rpc rollSpellDamage(rollSpellDamagemsg) returns (rollReply) {}
}

service rollAttackDamage {
	rpc rollAttackDamage(rollAttackDamagemsg) returns (rollReply) {}
}

service rollSave {
	rpc rollSave(rollSavemsg) returns (rollReply) {}
}

message rollReply{
	string roll_summary = 1;
	int32 roll_total = 2;
	int32 roll_total_modified = 3;
	string user = 4;
	string channel = 5;
}

message rollAttackmsg {
	bool adv = 1;
	bool dadv = 2;
	string weapon = 3;
	string user = 4;
	string channel = 5;
}

message rollIniativemsg {
	bool adv = 1;
	bool dadv = 2;
	string user = 3;
	string channel = 4;
}

message rollSpellcastmsg {
	bool adv = 1;
	bool dadv = 2;
	string spell = 3;
	string user = 4;
	string channel = 5;
}

message rollAttackDamagemsg {
	bool adv = 1;
	bool dadv = 2;
	string weapon = 3;
	string user = 4;
	string channel = 5;
}

message rollSpellDamagemsg {
	bool adv = 1;
	bool dadv = 2;
	string spell = 3;
	string user = 4;
	string channel = 5;
}

message rollSavemsg {
	bool adv = 1;
	bool dadv = 2;
	string stat = 3;
	string user = 4;
	string channel = 5;
}
